src/ios-plugin/swift-lang/src/main/java/io/ecocode/swift/antlr/ParseTreeAnalyzer.java

```diff
@@ -0,0 +1,54 @@
+package io.ecocode.swift.antlr;
+
+import io.ecocode.antlr.AntlrContext;
+import io.ecocode.antlr.ParseTreeItemVisitor;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+public class ParseTreeAnalyzer {
+
+    private static final int EXECUTOR_TIMEOUT = 10000;
+
+    private static final Logger LOGGER = Loggers.get(ParseTreeAnalyzer.class);
+
+    private final String languageKey;
+    private final Type type;
+    private final AntlrContext antlrContext;
+    private final SensorContext sensorContext;
+
+    public ParseTreeAnalyzer(String languageKey, Type type, AntlrContext antlrContext, SensorContext sensorContext) {
+        this.languageKey = languageKey;
+        this.type = type;
+        this.antlrContext = antlrContext;
+        this.sensorContext = sensorContext;
+    }
+
+    public void analyze(final ParseTreeItemVisitor... visitors) {
+
+        FilePredicate hasLang = sensorContext.fileSystem().predicates().hasLanguage(languageKey);
+        FilePredicate hasType = sensorContext.fileSystem().predicates().hasType(type);
+        FilePredicate langAndType = sensorContext.fileSystem().predicates().and(hasLang, hasType);
+        final Charset charset = sensorContext.fileSystem().encoding();
+
+        for (InputFile inf : sensorContext.fileSystem().inputFiles(langAndType)) {
+
+            // Visit source files
+            try {
```

Comment:
maybe use try-with-resource to auto-close files

Reply:
There is actually nothing to autoclose here: `antlrContext.loadFromFile(inf, charset)` is the only line throwing)

Reply:
ok